client/signup/utils.js

```diff
@@ -216,3 +217,22 @@ export const shouldForceLogin = ( flowName, userLoggedIn ) => {
 	const flow = flows.getFlow( flowName, userLoggedIn );
 	return !! flow && flow.forceLogin;
 };
+
+// Derive if the "plans" step actually will be visible to the customer in a given flow after the domain step
+// i.e. Check "launch-site" flow while having a purchased paid plan
+export const isPlanSelectionAvailableLaterInFlow = ( flowSteps, isPlanStepSkipped ) => {
+	/**
+	 * Caveat here even though "plans" step maybe available in a flow it might not be active
+	 * i.e. Check flow "domain"
+	 */
+
+	const plansIndex = flowSteps.findIndex(
+		( stepName ) => getStepModuleName( stepName ) === 'plans'
+	);
+	const domainsIndex = flowSteps.findIndex(
+		( stepName ) => getStepModuleName( stepName ) === 'domains'
+	);
+	const isPlansStepExistsInFutureOfFlow = plansIndex > 0 && plansIndex > domainsIndex;
+
+	return isPlansStepExistsInFutureOfFlow && ! isPlanStepSkipped;
```

Comment:
Might be out of scope, but I'd check for `isPlanStepSkipped` in the beginning of the function and bail out early without running these loops.

Reply:
Thanks for dropping this note. Actually, this pointed out that this function was doing more than one thing. See https://github.com/Automattic/wp-calypso/pull/54117/commits/4e61585670df95bdf0d42fa36a5ca938e45a51e4 with a simplification.